maven-core/src/main/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactory.java

```diff
@@ -238,9 +263,83 @@ else if ( request.isUpdateSnapshots() )
+        }
         return session;
     }
 
+    private FileTransformerManager newFileTransformerManager()
+    {
+        return new FileTransformerManager()
+        {
+            @Override
+            public Collection<FileTransformer> getTransformersForArtifact( final Artifact artifact )
+            {
+                Collection<FileTransformer> transformers = new ArrayList<>();
+                if ( "pom".equals( artifact.getExtension() ) )
+                {
+                    final TransformerFactory transformerFactory = TransformerFactory.newInstance();
+
+                    transformers.add( new FileTransformer()
+                    {
+                        @Override
+                        public InputStream transformData( File file )
+                            throws IOException, TransformException
+                        {
+                            final PipedOutputStream pipedOutputStream  = new PipedOutputStream();
+                            final PipedInputStream pipedInputStream  = new PipedInputStream( pipedOutputStream );
+                            
+                            final SAXSource transformSource;
+                            try
+                            {
+                                transformSource =
+                                    new SAXSource( consumerPomXMLFilterFactory.get().get( file.toPath() ),
+                                                   new InputSource( new FileReader( file ) ) );
+                            }
+                            catch ( SAXException | ParserConfigurationException e )
+                            {   
+                                e.printStackTrace();
+                                throw new TransformException( "Failed to create a consumerPomXMLFilter", e );
+                            }
+                            
+                            final StreamResult result = new StreamResult( pipedOutputStream );
+                            
+                            final Runnable runnable = new Runnable()
+                            {
+                                @Override
+                                public void run()
+                                {
+                                    try ( PipedOutputStream out = pipedOutputStream )
+                                    {
+                                        transformerFactory.newTransformer().transform( transformSource, result );
+                                    }
+                                    catch ( TransformerException | IOException e )
+                                    {
+                                        e.printStackTrace();
+                                        throw new RuntimeException( e );
+                                    }
+                                }
+                            };
+
+                            new Thread( runnable ).start();
```

Comment:
maybe it is better to give a name to this thread and consider make it a daemon thread.
I don't know if it is worth to use a thread pool, in order to save the creation/destruction of a thread for each pom to transform

Reply:
@rfscholte 
why so complitaed? The J8 would make your code nicely compact.
So. It does not make sense to have annonymous classes and large blocks of code.
Simply type `new Thread( () -> { ... } )` and then call `setDaemon( true )` , `setName( "..." )`... and start it up!
That's the reason and the entire code in Maven would be better readable and smaller!

Reply:
@eolivelli 
@rfscholte 
One more question.
Why if there a Thread, there's not thread pool executor instance and a configuration via CLI args? We had the thread pool in the resolver and it also could be configured. It would make sense for me to let the user to have control over the thread resources.

Reply:
(Tibor ;)) Please dont use daemon threads, just handle their lifecycle with a listener otherwise you can create leaks easily if  a transformer starts to hang

Reply:
To close this discussion: Code is based on examples for PipedStreams (requirement for me to lower memoryconsumption), but they still use Threads. I'll rewrite them to Callables

Reply:
@rmannibucau 
Daemon threads are importand to properly exit the application when it is about to exit; otherwise the application means Maven will hang.

Reply:
The advices I got:
> 1. I would not use ExecutorService. Instead, I would start a single 
> thread and give it a unique name, set it to Daemon status.
> 
> 2. The exceptions are not managed well. If an IOException occurs in your 
> solution, you wouldn't know about it.
I'd suggest to remember this as improvements once somebody hits an issue. So far I haven't seen the need to change it.